### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by TitiAlone**
This is a retake on @arobase-che's previous work on updating dependencies for zmarkdown packages.

## List of changed behaviours

Here's a list of what has changed (and hence, what needs to be tested on zds-site) for reference, made by looking at the modified snapshots and fixtures:

- a blank line now separate blockquotes (solves #307);
- use \footnote instead of list of links for references in LaTeX;
- empty code spans do not render anything anymore;
- KaTeX SVG has the `xmlns` attribute (solves one half of #345);
- lists with empty items render a list instead of nothing;
- tables now take precedence over code blocks;
- footnotes are wrapped into a paragraph when outside a list;
- footnotes are not anymore wrapped into a paragraph when part of a list;
- footnotes order has changed ([see below](https://github.com/zestedesavoir/zmarkdown/pull/371#discussion_r373767234));
- generally, paragraphs are not added inside lists anymore;
- hl_lines and linenostart parameters for code blocks are back again **for LaTeX only** (solves one half #255 and #340);
- captions now break lists (this was expected by a test, which was incorrect before);
- quite a lot of changes are expected from KaTeX, please report any issue found;
- by the way, KaTeX was updated to version 0.11.1, so please update KaTeX also on zds-site (to 0.11.1) when deploying this version of zmarkdown.

## What remains?

First of all, I would like a global approval of the changes above by @artragis .
Maybe a review is needed (poke @vhf ), but I honestly don't know how this PR could be checked, maybe by looking at the updated tests.
Most importantly, it would have been great to do a rebase, but this is very complicated to do correctly here, so I think it will be a simple merge.
Finally, dependencies of `remark-image-download` were NOT updated, because they would need huge refactoring of the code, and I prefer to merge this first before attempting to do it (changes are less urgent, because they don't concern security issues nor are they feature-related).

### CODE REVIEW

**Comment by TitiAlone** [comment]
Now with rebase, please check but I think it is okay; I had to do a commit because some snapshots were still in their old version after the rebase.

File: packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap [hunk 1]
```diff
@@ -5,18 +5,9 @@ exports[`{"gfm":false,"commonmark":false,"footnotes":true} footnote-split 1`] =
...
-<p>third def</p>
-<a href=\\"#fnref-third-postfix\\" class=\\"footnote-backref\\">↩</a>
-</li>
-<li id=\\"fn-1-postfix\\">
-<p>second def</p>
-<a href=\\"#fnref-1-postfix\\" class=\\"footnote-backref\\">↩</a>
-</li>
+<li id=\\"fn-first-postfix\\">first def<a href=\\"#fnref-first-postfix\\" class=\\"footnote-backref\\">↩</a></li>
+<li id=\\"fn-1-postfix\\">second def<a href=\\"#fnref-1-postfix\\" class=\\"footnote-backref\\">↩</a></li>
+<li id=\\"fn-third-postfix\\">third def<a href=\\"#fnref-third-postfix\\" class=\\"footnote-backref\\">↩</a></li>
```
Comment (vhf) [inline packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: Corresponding fixture:

```md
a[^first]b[^second def]c[^third]

[^first]: first def
[^third]: third def
```

The behavior changed, probably for the better.

File: packages/remark-captions/__tests__/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -21,9 +21,10 @@ print('bla')
 exports[`captions should break lists 1`] = `
 "<figure><blockquote>
 <ul>
-<li>Bar
-Source: fc2</li>
+<li>Bar</li>
 </ul>
+</blockquote><figcaption>fc2</figcaption></figure>
+<figure><blockquote>
```
Comment (vhf) [inline packages/remark-captions/__tests__/__snapshots__/index.js.snap]: Nice side effect of the new "blank lines between quotes create separate quotes" behavior.

File: packages/remark-grid-tables/package.json [hunk 3]
```diff
@@ -33,8 +33,9 @@
   "license": "MIT",
   "dependencies": {
     "grapheme-splitter": "^1.0.4",
+    "is-fullwidth-code-point": "^3.0.0",
```
Comment (vhf) [inline packages/remark-grid-tables/package.json]: Extraneous dep: not actually used in this package. That's just a tiny merge mistake.

File: packages/remark-images-download/src/index.js [hunk 4]
```diff
@@ -1,3 +1,6 @@
+require('core-js/stable')
+require('regenerator-runtime/runtime')
```
Comment (vhf) [inline packages/remark-images-download/src/index.js]: Both of these seem overkill. Upgrading this package independently as you suggested makes sense IMO, I'd suggest reverting this file and only bumping unist-util-visit/rimraf/shortid for now.

File: packages/zmarkdown/__tests__/__snapshots__/latex.tests.js.snap [hunk 5]
```diff
@@ -285,7 +290,7 @@ exports[`list 1`] = `
 `;
 
 exports[`math 1`] = `
-"A sentence ($S$) with \\\\textit{italic} and inline math ($C_L$) and $$b$$ another.
+"A sentence ($S$) with \\\\textit{italic} and inline math ($C_L$) and $b$ another.
```
Comment (vhf) [inline packages/zmarkdown/__tests__/__snapshots__/latex.tests.js.snap]: Not sure where this change comes from and if it's intentional.

File: packages/zmarkdown/__tests__/new-html-suite.test.js [hunk 6]
```diff
@@ -137,7 +137,7 @@ describe('tables', () => {
       \`ici ok\`| \`gauche | droite\`
     `
 
-    return expect(renderString(input)).resolves.toContain('<td><code>gauche | droite</code></td>')
+    return expect(renderString(input)).resolves.toContain('<td>`gauche</td>')
```
Comment (vhf) [inline packages/zmarkdown/__tests__/new-html-suite.test.js]: I don't think that's correct, the last cell should contain `<code>gauche | droite</code>`:

```md
Titre 1 | Titre 2
--------|--------
`ici ok`| `gauche | droite`
```

Comment (TitiAlone) [reply packages/zmarkdown/__tests__/new-html-suite.test.js]: I thought the same, so I opened an issue on [remark](https://github.com/remarkjs/remark/issues/465), and this is actually the expected behaviour.

File: packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap (hunk 1 above)
Comment (TitiAlone) [reply packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: I don't know what you think changed, but from what I can see, only the order of the three footnotes changed.

File: packages/remark-captions/__tests__/__snapshots__/index.js.snap (hunk 2 above)
Comment (TitiAlone) [reply packages/remark-captions/__tests__/__snapshots__/index.js.snap]: I had a doubt on this one, but the title "captions should break lists" led me into thinking that the current behaviour was the expected one.

File: packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap (hunk 1 above)
Comment (vhf) [reply packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: Yeah I was talking about the order in which footnotes are rendered. Footnotes are displayed in the same order as their definition.

```md
a[^first_footnote_reference]

b[^`b` second footnote reference but first footnote definition]

c[^last_footnote_reference]

[^last_footnote_reference]: `c` last footnote reference but second footnote definition
[^first_footnote_reference]: `a` first footnote reference but last footnote definition
```

Before:

image

After:

image

In the HTML ordered list, the list items 1/2/3 don't match the footnote references numbers.

Comment (TitiAlone) [reply packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: Ok, I see what you mean, added it to the issue status to get an approval of the change.

Comment (artragis) [reply packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: this cannot be good : if you click on "3" you get redirected to "c" whereas it should lead you to "a"

Comment (vhf) [reply packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: Yeah, the footnote reference (superscript) and the list item number should match.

a^3, 3. a

File: packages/zmarkdown/__tests__/__snapshots__/latex.tests.js.snap (hunk 5 above)
Comment (TitiAlone) [reply packages/zmarkdown/__tests__/__snapshots__/latex.tests.js.snap]: You're right, this is a bad behaviour, finally found it and solved it!

File: packages/zmarkdown/config/remark.js [hunk 7]
```diff
@@ -317,10 +317,6 @@ const remarkConfig = {
     inlineMathDouble: true,
   },
 
-  katex: {
-    inlineMathDoubleDisplay: true,
```
Comment (artragis) [inline packages/zmarkdown/config/remark.js]: pourquoi retirer celui-là?

Comment (TitiAlone) [reply packages/zmarkdown/config/remark.js]: It doesn't exist anymore, the behaviour targeted by this option is now the default; rehype-katex decided to use remark-math's `inlineMathDouble` option (indirectly) to detect the option instead.

File: packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap (hunk 1 above)
Comment (TitiAlone) [reply packages/rehype-postfix-footnote-anchors/__tests__/__snapshots__/index.js.snap]: This is fixed in zmarkdown; the behaviour is normal in rehype, so the test will not be updated for the package.

**Comment by artragis** [comment]
> captions now break lists (this was expected by a test, which was incorrect before);

can you explain this, And why not openning an issue that will be closed by this PR so that we can track the change?

> by the way, KaTeX was updated to version 0.11.1, so please update KaTeX also on zds-site (to 0.11.1) when deploying this version of zmarkdown.

Having a look at https://github.com/KaTeX/KaTeX/releases, I see 

> Fix \sqrt SVG path

so @Situphen we will clearly need to update katex dependency on zds-site side

**Comment by TitiAlone** [comment]
> can you explain this, And why not openning an issue that will be closed by this PR so that we can track the change?

Please refer to [this test](https://github.com/zestedesavoir/zmarkdown/pull/371/files#diff-b029b179a6eb4d9d5394b1e79f925735). The new behaviour is expected, not a bug, but could maybe be update if it doesn't suit zds-site needs.

> so @Situphen we will clearly need to update katex dependency on zds-site side

Yes.

**Comment by vhf** [comment]
> > captions now break lists (this was expected by a test, which was incorrect before);
> 
> can you explain this

This [used to](https://zestedesavoir.github.io/zmarkdown/) generate a single quote:
```md
> Foo

> Bar
```

Now that it generates two quotes, this should be two quotes and their respective captions:
quote:
```md
> Foo
Source: foo

> Bar
Source: foo
```

Same goes for:
```md
> * Foo
Source: foo

> * Bar
Source: foo
```

**Comment by artragis** [comment]
Ok so everything seems to be fine for me.

**Pull request merged**